nss-day-cohort-52/bangazon-api-luciidlou
# can now successfully favorite/unfavorite stores, as well as like/unli…

…ke products

### Issue ticket(s): #11 #12 

### New file(s): 
- `models/like.py`
- Added a new migration. This PR will require a new migration of the tables in order to test correctly

### Modified File(s): 
- `models/product.py` 
- `serializers/product_serializer.py` 
- `views/product_view.py` 
- `views/store_view.py`

### Modifications:
- Added a new `Like` model to the models directory
- Added a `likes` ManyToManyField on the `Product` model
- Added an `is_liked` BooleanField on the `ProductSerializer`
- Added 3 new custom actions to the `ProductView` class...
-- `like` performs a post request to create a `Like` object
-- `unlike` performs a delete request to delete a `Like` object
-- `liked` performs a get request to retrieve only the products that are liked by the current user
- Added 2 new custom actions to the `StoreView` class...
-- `favorite` performs a post request to create a `Favorite` object
-- `unfavorite` performs a delete request on a `Favorite` object

### Steps to test:
1. While in the terminal, use git fetch --all in the root directory of this repo
2. Checkout out to `al-favStores-likeProducts`
3. Be sure to make your migrations now since we have altered the structure of the tables
4. Start the server
5. Open Postman and execute a post request @ `http://localhost:8000/api/stores/2/favorite` to favorite the store with a pk of 2, you should recieve a message in the response that says `Store has been added as a favorite`
6. Now perform delete request on that Like object you just created @ `http://localhost:8000/api/stores/2/unfavorite` and you should now recieve a message in the response saying `Store has been removed from favorites`
7. Next, perform a post request @ `http://localhost:8000/api/products/2/like`, the response should be `product added to likes`
8. Go ahead an execute a get request @ `http://localhost:8000/api/products/liked`, you should see the store you liked in the response
9. Now we need to try and delete that Like object we created, so we can execute this request @ `http://localhost:8000/api/products/2/unlike` and our response should be `product removed from likes`
10. Finally, to make sure that our `liked` custom action is working properly, go ahead and perform another get @ `http://localhost:8000/api/products/liked`. The response should no longer contain the store that was unfavorited in the previous step.
